### PR TITLE
obs-ffmpeg: Add option to disable settings logging

### PR DESCRIPTION
**Pull request by CodeYan01**
Allows plugins to disable the logging of settings every time the ffmpeg source is updated, by passing "log_changes" as true in the settings.

### Description

Adds "log_changes" to the ffmpeg source settings. If false, it will not log changes every time the settings of the media source is updated. This does not affect logging of errors and warnings. I did not add this to the properties window because I assume it's used in support channels to check the settings of media sources, so it would be bad if users can set this directly.

### Motivation and Context

I am writing a playlist plugin that uses media sources internally, just as the Image Slideshow source uses Image sources internally. However, Media source spams the log when the settings are updated, which is not ideal for this sort of plugin. Image Slideshow does not have this problem because Image sources do not log their settings.

### How Has This Been Tested?

I made my plugin pass "log_changes" as false to see that it disables logging when the settings of the internal media source are modified. I also verified that Media Sources still log as usual, as users can't set the property.

### Types of changes

 - Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/obs-ffmpeg/obs-ffmpeg-source.c [hunk 1]
```diff
@@ -233,6 +235,11 @@ static obs_properties_t *ffmpeg_source_getproperties(void *data)
 static void dump_source_info(struct ffmpeg_source *s, const char *input,
 			     const char *input_format)
 {
+	if (!s->log_changes) {
+		UNUSED_PARAMETER(input);
+		UNUSED_PARAMETER(input_format);
+		return;
+	}
```
Comment (norihiro) [inline plugins/obs-ffmpeg/obs-ffmpeg-source.c:242]:
```suggestion
	if (!s->log_changes)
		return;
```
`UNUSED_PARAMETER` is a macro to suppress a warning that the parameter is not used in the function. We don't need to have this in the `if` block.

Comment (CodeYan01) [reply plugins/obs-ffmpeg/obs-ffmpeg-source.c]: Resolved, thank you!

**Comment by derrod** [comment]
There's a merge conflict, please resolve it. Generally this seems fine.

**Comment by Lain-B** [comment]
Although this is a very niche feature, because it affects your plugin's users, and because it may be useful for us in the future, I'll merge this.

**Comment by CodeYan01** [comment]
Thank you very much, and I apologize for not seeing the merge conflict earlier

**Pull request merged**